teps/0094-step-resources-at-runtime.md

```diff
@@ -0,0 +1,61 @@
+---
+status: proposed
+title: Step Resources at Runtime
+creation-date: '2021-11-08'
+last-updated: '2021-11-08'
+authors:
+- '@lbernick'
+---
+
+# TEP-0094: Step Resources at Runtime
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Requirements](#requirements)
+<!-- /toc -->
+
+## Summary
+Add runtime configuration options for setting resource requirements of `Step`s.
+
+Currently, users can specify resource requirements in a `Task` definition,
+via the `Resources` field of each `Step`. However, there is currently no support
+for modifying these requirements in a `TaskRun` or `PipelineTaskRun`.
+
+This TEP proposes adding a configuration option to `TaskRun` and `PipelineTaskRun`
+to override any `Step` resource requirements specified in a `Task` or `PipelineTask`.
```

Comment:
Also SideCars -- the specific case we had where we had a problem was for a dind sidecar that needed more memory